src/cryptsetup/cryptsetup-generator.c

```diff
@@ -106,6 +106,38 @@ static int generate_keydev_mount(const char *name, const char *keydev, char **un
         return 0;
 }
 
+static int print_dependencies(FILE *f, const char* device_path) {
+        if (PATH_IN_SET(device_path, "/dev/urandom", "/dev/random", "/dev/hw_random"))
+                fputs("After=systemd-random-seed.service\n", f);
+        else if (!STR_IN_SET(device_path, "-", "none")) {
+                _cleanup_free_ char *udev_node;
+
+                udev_node = fstab_node_to_udev_node(device_path);
+                if (!udev_node)
+                        return log_oom();
+
+                if (!path_equal(udev_node, "/dev/null")) {
+                        if (path_startswith(udev_node, "/dev/")) {
```

Comment:
Hmm, maybe do
```C
  if (path_equal(udev_node, "/dev/null"))
           return 0;
```
here instead? This will reduce the indentation a bit and make this easier to read.